apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/csv-data-table.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  IconButton,
+  Portal,
+  Select,
+  Table,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@chakra-ui/react";
+import {
+  ChevronFirstIcon,
+  ChevronLastIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+} from "lucide-react";
+import { type Column, usePagination, useTable } from "react-table";
+import { Text } from "tw-components";
+
+interface CsvDataTableProps<T extends object> {
+  data: T[];
+  portalRef: React.RefObject<HTMLDivElement | null>;
+  columns: Column<T>[];
+}
+/**
+ * Display the data uploaded from useCsvUpload, using react-table
+ */
+export function CsvDataTable<T extends object>({
+  data,
+  portalRef,
+  columns,
+}: CsvDataTableProps<T>) {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    prepareRow,
+    // Instead of using 'rows', we'll use page,
+    page,
+    // which has only the rows for the active page
+    // The rest of these things are super handy, too ;)
+    canPreviousPage,
+    canNextPage,
+    pageOptions,
+    pageCount,
+    gotoPage,
+    nextPage,
+    previousPage,
+    setPageSize,
+    state: { pageIndex, pageSize },
+  } = useTable(
```

Comment:
if we're updating this component lets please update this to [tanstack table](https://tanstack.com/table/latest/docs/introduction) while we're at it

Reply:
Sure. I'll stack up the PRs